src/backend/graphics/opengl.rs

```diff
@@ -0,0 +1,17 @@
+use nix::c_void;
+
+pub enum SwapBuffersError {
+    ContextLost,
+    AlreadySwapped,
+}
+
+pub trait OpenGLRenderer
+{
+    fn swap_buffers(&self) -> Result<(), SwapBuffersError>;
+    unsafe fn get_proc_address(&self, symbol: &str) -> *const c_void;
+    fn get_framebuffer_dimensions(&self) -> (u32, u32);
+    fn is_current(&self) -> bool;
+    unsafe fn make_current(&self);
+    fn get_api(&self) -> Api;
+    fn get_pixel_format(&self) -> PixelFormat;
```

Comment:
`gluim` does not need `get_api` or `get_pixel_format`, actually no real library needs it, `glutin` just provides both of them. `glx` does not even need any of these, except `get_proc_address`. It just depends on the gl context being the *current* one and also expects that not to change over time and figures out the rest itself via low-level c calls internally. However anybody using raw GL might want functionally such as `get_api` to ease development.
I am just not happy with how directly linked this backend is to `glium`s requirements. It does not feel very generic or useful, if you just copy types and traits from `glutin`.

However you are right, it should be quite easy to put this over any backend type, we are currently aiming at. And I feel like it is already very complete.